src/app/components/core/control-dispatcher/pipelines/add-pipeline/add-control-pipeline.component.ts

```diff
@@ -0,0 +1,682 @@
+import { forkJoin, of } from 'rxjs';
+import { catchError, map } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-add-control-pipeline',
+  templateUrl: './add-control-pipeline.component.html',
+  styleUrls: ['./add-control-pipeline.component.css'],
+})
+export class AddControlPipelineComponent implements OnInit {
+  @ViewChild('pipelineForm') pipelineForm: NgForm;
+  @ViewChild(FilterAlertComponent) filterAlert: FilterAlertComponent;
+  @ViewChild('filterConfigComponent') filterConfigComponent: ConfigurationGroupComponent;
+
+  pipelines = [{ name: 'None' }];
+  selectedExecution = '';
+  selectedSourceType = {cpsid: null, name: ''};
+  selectedSourceName = '';
+  selectedDestinationType = {cpdid: null, name: ''};
+  selectedDestinationName = '';
+  public isPipelineEnabled = true;
+  QUOTATION_VALIDATION_PATTERN = QUOTATION_VALIDATION_PATTERN;
+  sourceTypeList = [];
+  destinationTypeList = [];
+  sourceNameList = [];
+  destinationNameList = [];
+  editMode = false;
+  pipelineID : number;
+  pipelineName;
+
+  public filterPipeline = [];
+  public isFilterOrderChanged = false;
+  public filterConfiguration: any;
+  filterConfigurationCopy: any;
+  public selectedFilterPlugin;
+  changedFilterConfig: any;
+  public deletedFilterPipeline = [];
+  public isFilterDeleted = false;
+  confirmationDialogData = {};
+  public isAddFilterWizard;
+  // To hold API calls to execute
+  apiCallsStack = [];
+  changedConfig = {};
+  
+  constructor(
+    private cdRef: ChangeDetectorRef,
+    private assetService: AssetsService,
+    private route: ActivatedRoute,
+    private controlPipelinesService: ControlPipelinesService,
+    private alertService: AlertService,
+    private ngProgress: ProgressBarService,
+    public rolesService: RolesService,
+    private dialogService: DialogService,
+    private response: ResponseHandler,
+    public sharedService: SharedService,
+    private schedulesService: SchedulesService,
+    private controlService: ControlDispatcherService,
+    public notificationService: NotificationsService,
+    private configService: ConfigurationService,
+    private filterService: FilterService,
+    private docService: DocService,
+    private fileUploaderService: FileUploaderService,
+    private configurationControlService: ConfigurationControlService,
+    private router: Router) { }
+
+  ngOnInit(): void {
+    this.getSourceDestTypes('source');
+    this.getSourceDestTypes('destination');
+    this.route.params.subscribe(params => {
+      this.pipelineID = params['id'];
+      if (this.pipelineID) {
+        this.editMode = true;
+        this.getControlPipeline();
+      } else {
+        this.selectedExecution = 'Shared';
+        this.selectedSourceType = {cpsid: 1, name: "Any"};
+        this.selectedDestinationType = {cpdid: 4, name: "Broadcast"};
+      }
+    }); 
+  }
+
+  ngAfterContentChecked(): void {
+    this.cdRef.detectChanges();
+  }
+
+  refresh() {
```

Comment:
what if refresh clicked when no this.editMode

Reply:
In refresh(), method we call getControlPipeline(). In getControlPipeline() method, we need pipeline ID to make call to get any pipeline. So there will be error if we click refresh when no this.editMode.
Btw; I have removed (in the next commit) refresh method as we are calling only one method (i.e. getControlPipeline()) only under this. So I made call directly to it.